scvi/external/cellassign/_model.py

```diff
@@ -0,0 +1,102 @@
+import logging
+
+import numpy as np
+import pandas as pd
+import torch
+
+from anndata import AnnData
+
+from scvi.data import register_tensor_from_anndata
+from scvi.dataloaders import AnnDataLoader
+from scvi.external.cellassign._module import CellAssignModule
+from scvi.lightning import TrainingPlan
+from scvi.model.base import BaseModelClass
+
+logger = logging.getLogger(__name__)
+
+
+class CellAssign(BaseModelClass):
+    """
+    Reimplementation of CellAssign for reference-based annotation [Zhang19]_.
+
+    Parameters
+    ----------
+    adata
+        single-cell AnnData object that has been registered via :func:`~scvi.data.setup_anndata`.
+    cell_type_markers
+        Binary marker gene matrix
+    use_gpu
+        Use the GPU or not.
+    **model_kwargs
+        Keyword args for :class:`~scvi.external.CellAssignModels`
+
+    Examples
+    --------
+    >>> # TODO
```

Comment:
add an example like in the notebook we were looking at today.

1. load data, marker matrix
2. initialize model
3. train
4. predict

see our other docstrings for examples